iphone/Maps/UI/PlacePage/Components/ElevationProfile/ElevationProfilePresenter.swift

```diff
@@ -56,30 +57,37 @@ final class ElevationProfilePresenter: NSObject {
       DescriptionsViewModel(title: L("elevation_profile_min_elevation"), value: trackInfo.minElevation, imageName: "ic_em_min_attitude_24")
     ]
   }
-
-  deinit {
-    bookmarkManager.resetElevationActivePointChanged()
-    bookmarkManager.resetElevationMyPositionChanged()
-  }
 }
 
 extension ElevationProfilePresenter: ElevationProfilePresenterProtocol {
-  func update(trackInfo: TrackInfo, profileData: ElevationProfileData?) {
-    self.profileData = profileData
-    if let profileData {
+  func update(with trackData: PlacePageTrackData) {
+    self.trackData = trackData
+    if let profileData = trackData.elevationProfileData {
       self.chartData = ElevationProfileChartData(profileData)
     } else {
       self.chartData = nil
     }
-    descriptionModels = Self.descriptionModels(for: trackInfo)
+    descriptionModels = Self.descriptionModels(for: trackData.trackInfo)
     configure()
   }
 
+  func updateActivePoint(_ distance: Double) {
+    guard let view, !view.isChartViewInfoHidden else { return }
+    view.setActivePoint(distance)
+  }
+
+  func updateMyPosition(_ distance: Double) {
+    guard let view, !view.isChartViewInfoHidden else { return }
+    view.setMyPosition(distance)
+  }
+
   func configure() {
     view?.isChartViewHidden = false
 
     let kMinPointsToDraw = 3
-    guard let profileData, let chartData, chartData.points.count >= kMinPointsToDraw else {
+    guard let profileData = trackData.elevationProfileData,
+          let chartData,
+          chartData.points.count >= kMinPointsToDraw else {
```

Comment:
Why a single point can't be drawn? It is already useful to see the current altitude/time on the graph.